File: 0541-reverse-string-ii/0541-reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        StringBuilder sb = new StringBuilder();

        int i = 0, j = 0;
        
        while (i < s.length()) {
            
            j = Math.min(i + k, s.length());
            sb.append((new StringBuilder(s.substring(i, j))).reverse());

            if (j >= s.length()) {
                break;
            }

            i = j;
            j = Math.min(i + k, s.length());
            sb.append(s, i, j);

            i = j;
        }

        return sb.toString();
    }
}

File: 0541-reverse-string-ii/0541-reverse-string-ii.kt
class Solution {
fun reverseStr(s: String, k: Int): String {
    var str = s
    var i = 0
    while (i < str.length) {
        if (i + k < str.length) {
            str = str.substring(0, i) + str.substring(i, i + k).reversed() + str.substring(i + k)
        } else {
            str = str.substring(0, i) + str.substring(i).reversed()
        }
        i += 2 * k
    }
    return str
}
}